src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -337,11 +345,14 @@ internal sealed class SrCnnEntireModeler
             private double[] _trends;
             private double[] _curWindow;
 
-            public SrCnnEntireModeler(double threshold, double sensitivity, SrCnnDetectMode detectMode)
+            private int _period;
```

Comment:
>private int _period; [](start = 12, length = 20)

this can be readonly

Reply:
Also, please put it next to the other readonly fields `_threshold`, `_sensitivity`, `_detectMode`.

---
In reply to: [435116553](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435116553) [](ancestors = 435116553)

Reply:
updated.

---
In reply to: [435215209](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435215209) [](ancestors = 435215209,435116553)